Gems/ROS2/Code/Include/ROS2/Communication/TopicConfiguration.h

```diff
@@ -21,6 +21,12 @@ namespace ROS2
         AZ_TYPE_INFO(TopicConfiguration, "{7535D58F-5284-4657-A799-1F69D3F5AA42}");
         static void Reflect(AZ::ReflectContext* context);
 
+        TopicConfiguration() = default;
+        TopicConfiguration(const QoS& qos)
+        {
+            m_qos = qos;
+        }
```

Comment:
nit: you can also do this
```suggestion
        TopicConfiguration(const QoS& qos) :
            m_qos(qos)
        {
        }
```